### ISSUE TITLE
Edit Modes without Folders

### ISSUE THREAD

**Issue by OrionSR** [sannybuilder/dev#16]
Since the suggestions for new entries for opcode.txt and gxt got a favorable priority, only a few more steps are necessary in order to make edit modes completely contained within a <game> definition in games.xml. I've been trying to account for all-the-rest found in the data folder for the edit mode.

GXT and opcode.txt location are currently under consideration.

DAT and IDE files can be managed with <data>. Possibly, look for default.dat and gta3.dat in the usual file structure if custom*.dat isn't specified.

Need <binary> for managing only 4 files that don't need to change or propagate.

What else? I recently realized that Make opcode.txt tool writes to the data folder. This is fairly destructive behavior in the context of multiple edit modes sharing the same resources. For sa_mobile type edit modes the install directory might be a more appropriate destination.

**Comment by x87** [comment]
I changed the behavior for the Make opcode.txt to write to the same destination as specified with `<examples>` entry in the games.xml #5 In this case an opcodes.txt could have put into completely unique directories for each mode.

For IDE files I think creating another entry in the games.xml to specifiy which IDE/DAT files to load should be a fairly reasonable approach:

```xml
    <game shortcut="GTA3" title="GTA III" mode="gta3" type="default">
        <ide>@game:\data\default.ide</ide>
    </game>
    <game shortcut="VC" title="GTA VC" mode="vc" type="default">
        <ide>@game:\data\default.ide</ide>
    </game>
    <game shortcut="SA" title="GTA SA" mode="sa" type="default">
        <ide>@game:\data\default.dat</ide>
        <ide>@game:\data\maps\veh_mods\veh_mods.ide</ide>
    </game>
    <game shortcut="LCS" title="GTA LCS" mode="lcs" type="default">
        <ide>@sb:\data\lcs\models.ide</ide>
    </game>
    <game shortcut="VCS" title="GTA VCS" mode="vcs" type="default">
        <ide>@sb:\data\vcs\models.ide</ide>
    </game>
    <game shortcut="SA Mobile" title="SA Mobile" mode="sa_mobile" type="default">
        <ide>@sb:\data\sa_mobile\custom.dat</ide>
    </game>
```

**Comment by OrionSR** [comment]
I'm a big fan of the Make opcode.txt tool but, my experience as a new user was similar to:

"Hm... Wonder what this does."
![2020-02-09](https://user-images.githubusercontent.com/15115889/74112171-2d232900-4b4f-11ea-9aa0-41c169183da3.png)
"Sure, why not?" then, "Where'd my file go?" and later. "What's this do?"
![2020-02-09 (1)](https://user-images.githubusercontent.com/15115889/74112214-9b67eb80-4b4f-11ea-8bb1-ed9879304570.png)
Because I had a SaveAnywhere script open for Make opcode.txt.

Locking this power tool when not in admin mode seems a bit extreme. Most users won't ever need to use this tool. A lot more warning seems warranted. 

Maybe change the first dialog to **Open** (in Sanny), Save As (working directory), Overwrite, or Cancel, And then throw a clarified warning if Overwrite is selected.

**Comment by OrionSR** [comment]
The other issue is that I was planning to share the same opcode.txt and other command files for several modes. 
For example, PCv1 PCv2, PC_Steam, PC_Downgraded, just to cover what's currently installed.

The intended function is that I could update one and have it propagate to all, I just want this to be a very deliberate process.

**Comment by x87** [comment]
Those modes could have a path to the same opcodes.txt file, right?

**Comment by OrionSR** [comment]
Yes, that's the plan. With the addition of examples in games.xml I plan to group all command files for a particular game: SCM.ini, opcode.txt, keywords.txt, and classes.db. If I can get things set up the way I plan, the only difference between an edit mode using the traditional community script commands and an SCR format like fastman92's script command definitions will be a small change to a root folder of these commands.
